topics/go/algorithms/slices/max/max_test.go

```diff
@@ -1,31 +1,41 @@
 package slices_test
 
 import (
-	slices "github.com/ardanlabs/gotraining/topics/go/algorithms/slices/max"
 	"testing"
-)
 
-const succeed = "\u2713"
-const failed = "\u2717"
+	slices "github.com/ardanlabs/gotraining/topics/go/algorithms/slices/max"
+)
 
 func TestMax(t *testing.T) {
-
-	maxTests := []struct {
-		name     string
-		input    []int
-		expected int
+	tests := map[string]struct {
+		input       []int
+		expected    int
+		shouldError bool
 	}{
-		{"Slice with one element", []int{10}, 10},
-		{"Slice with even number of elements", []int{10, 30}, 30},
-		{"Slice with odd number of elements", []int{10, 50, 30}, 50},
+		"Empty Slice":                        {[]int{}, 0, true},
+		"nil Slice":                          {[]int(nil), 0, true},
+		"Slice with one element":             {[]int{10}, 10, false},
+		"Slice with even number of elements": {[]int{10, 30}, 30, false},
+		"Slice with odd number of elements":  {[]int{10, 50, 30}, 50, false},
 	}
 
-	for _, tt := range maxTests {
-		got := slices.Max(tt.input)
-		if got != tt.expected {
-			t.Logf("\t%s\t Incorrect value returned: %d\n.", failed, tt.input)
-			t.Fatalf("\t\tGot %d, Expected %d.", got, tt.expected)
-		}
-		t.Logf("\t%s\tCorrect value returend.", succeed)
+	t.Parallel()
```

Comment:
I believe this needs to be inside each test function you will call t.Run on.

For consistency, can we follow these guidelines for tests please.

https://github.com/ardanlabs/gotraining/blob/master/topics/go/algorithms/strings/reverse/reverse_test.go